packages/datadog-esbuild/index.js

```diff
@@ -249,34 +274,70 @@ ${build.initialOptions.banner.js}`
         : data.pkg
 
       // Read the content of the module file of interest
-      const fileCode = fs.readFileSync(args.path, 'utf8')
+      let contents
+
+      if (data.isESM) {
+        if (args.path.endsWith(ESM_INTERCEPTED_SUFFIX)) {
+          args.path = args.path.slice(0, -1 * ESM_INTERCEPTED_SUFFIX.length)
+
+          if (data.internal) {
+            args.path = args.path.slice(INTERNAL_ESM_INTERCEPTED_PREFIX.length)
+          }
+
+          interceptedESMModules.add(args.path)
+
+          const setters = await processModule({
+            path: args.path,
+            internal: data.internal,
+            context: { format: 'module' }
+          })
+
+          const iitmPath = require.resolve('import-in-the-middle/lib/register.js')
+          const toRegister = data.internal ? args.path : pathToFileURL(args.path)
+          contents = `
+import { register } from ${JSON.stringify(iitmPath)}
+import * as namespace from ${JSON.stringify(args.path)}
+// Mimic a Module object (https://tc39.es/ecma262/#sec-module-namespace-objects).
```

Comment:
```suggestion
          // Mimic a Module object (https://tc39.es/ecma262/#sec-module-namespace-objects).
          contents = `
import { register } from ${JSON.stringify(iitmPath)}
import * as namespace from ${JSON.stringify(args.path)}
```

Can you hoist the comment out? That way it won't potentially appear many times in the bundle.